fastchat/model/apply_delta.py

```diff
@@ -2,33 +2,69 @@
 Apply the delta weights on top of a base model.
 
 Usage:
-python3 -m fastchat.model.apply_delta --base ~/model_weights/llama-13b --target ~/model_weights/vicuna-13b --delta lmsys/vicuna-13b-delta-v1.1
+python3 -m fastchat.model.apply_delta --base ~/model_weights/llama-13b --target ~/model_weights/vicuna-13b --delta lmsys/vicuna-13b-delta
 """
 import argparse
 
 import torch
 from tqdm import tqdm
 from transformers import AutoTokenizer, AutoModelForCausalLM
-
+from transformers import LlamaTokenizer, AutoConfig
+import gc
+import shutil
 
 def apply_delta(base_model_path, target_model_path, delta_path):
-    print(f"Loading the base model from {base_model_path}")
-    base = AutoModelForCausalLM.from_pretrained(
-        base_model_path, torch_dtype=torch.float16, low_cpu_mem_usage=True)
-    base_tokenizer = AutoTokenizer.from_pretrained(
-        base_model_path, use_fast=False)
 
     print(f"Loading the delta from {delta_path}")
-    delta = AutoModelForCausalLM.from_pretrained(delta_path, torch_dtype=torch.float16, low_cpu_mem_usage=True)
+    delta_tokenizer = AutoTokenizer.from_pretrained(delta_path, use_fast=False)
+
+
+    DEFAULT_PAD_TOKEN = "[PAD]"
+    base_tokenizer = AutoTokenizer.from_pretrained(base_model_path, use_fast=False)
+    num_new_tokens = base_tokenizer.add_special_tokens(dict(pad_token=DEFAULT_PAD_TOKEN))
+    print("num_new_tokens: ",num_new_tokens)
+
+    delta_num = 0
+    delta_state_dict = torch.load(f"{delta_path}/pytorch_model-0000{str(delta_num%3+1)}-of-00003.bin", map_location=torch.device('cpu'))
+    config = AutoConfig.from_pretrained(base_model_path)
+
+    
 
     print("Applying the delta")
-    for name, param in tqdm(base.state_dict().items(), desc="Applying delta"):
-        assert name in delta.state_dict()
-        param.data += delta.state_dict()[name]
+    for i in range(42):
```

Comment:
Do not use a hardcoded number. Please
1. Ues the latest main branch of huggingface/transformers
2. Redo the llama weight conversion by following https://huggingface.co/docs/transformers/main/model_doc/llama
3. Test the following deltas
```
--delta lmsys/vicuna-7b-delta-v1.1
--delta lmsys/vicuna-13b-delta-v1.1
```